File: solutions/clojure/095-rotate-array.clj
;; Problem 95: Rotate Array (LeetCode 189)
;; Difficulty: Med
;; Language: Clojure
;; 
(defn rotate [nums k]
  (let [n (count nums) s (mod k n)]
    (concat (drop (- n s) nums) (take (- n s) nums))))

File: solutions/scala/095-rotate-array.scala
/**
 * Problem 95: Rotate Array (LeetCode 189)
 * Difficulty: Med
 * Language: Scala
 */
def rotate[A](nums: Vector[A], k: Int): Vector[A] = {
  val n = nums.length
  val s = k % n
  nums.drop(n - s) ++ nums.take(n - s)
}
